app.go

```diff
@@ -21,6 +26,41 @@ func NewApplication(l log.Logger, s stats.Statter) *Application {
 	}
 }
 
+func (a *Application) SaySomething2() AppResponseFunc {
```

Comment:
This is fairly complex for what it does. It could be replaced with somethings simple like:
```
func SaySomething2(loc l10n.LocaleInstance, user string) (ApplicationResponse, error) {
```

Reply:
Yeah, as we have a lambda function per intent anyway that makes total sense, I got a bit mislead by generalization :)
I will keep though the `ResponseFunc` to make it a bit more flexible as Session, Context and User are optional.

Reply:
You could also just make them pointers, but the `With*` is still fine.